<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CoreUI</class>
 <widget class="QMainWindow" name="CoreUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1298</width>
    <height>654</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenCV Face Recognition System - Core</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="faceRecognizerSettingsLogGroupBox">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>10</y>
      <width>441</width>
      <height>391</height>
     </rect>
    </property>
    <property name="title">
     <string>Face recognition settings</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <widget class="QGroupBox" name="debugGroupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>421</width>
       <height>161</height>
      </rect>
     </property>
     <property name="title">
      <string>Debug settings</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <widget class="QCheckBox" name="debugCheckBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>131</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Debug Mode</string>
      </property>
     </widget>
     <widget class="QLabel" name="debugLabel">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>120</y>
        <width>421</width>
        <height>41</height>
       </rect>
      </property>
      <property name="lineWidth">
       <number>3</number>
      </property>
      <property name="text">
       <string>Note: The above adjustments will only be triggered if the debug mode is turned on</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>281</width>
        <height>71</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="debugGridLayout">
       <item row="1" column="0">
        <widget class="QLabel" name="autoAlarmThresholdLabel">
         <property name="text">
          <string>Automatic alarm threshold:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="confidenceThresholdLabel">
         <property name="text">
          <string>Confidence threshold:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSlider" name="autoAlarmThresholdSlider">
         <property name="minimum">
          <number>50</number>
         </property>
         <property name="maximum">
          <number>80</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="pageStep">
          <number>5</number>
         </property>
         <property name="value">
          <number>65</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksAbove</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSlider" name="confidenceThresholdSlider">
         <property name="maximum">
          <number>80</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="pageStep">
          <number>5</number>
         </property>
         <property name="value">
          <number>50</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksAbove</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="tipsGroupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>220</y>
       <width>421</width>
       <height>171</height>
      </rect>
     </property>
     <property name="title">
      <string>Instructions for use</string>
     </property>
     <widget class="QLabel" name="autoAlarmTipLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>401</width>
        <height>81</height>
       </rect>
      </property>
      <property name="text">
       <string>② If the confidence level exceeds the automatic alarm threshold, an alarm signal will be triggered. By default, the system will alarm when the alarm signal triggers more than a certain count to reduce false alarms and avoid thread blocking.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="confidenceTipLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>401</width>
        <height>61</height>
       </rect>
      </property>
      <property name="text">
       <string>① The smaller the confidence threshold, the higher the accuracy requirement. After the debug mode is enabled, the system will output face recognition feedback in real time.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QCheckBox" name="equalizeHistCheckBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>25</y>
       <width>181</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Histogram equalization</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="systemSettingsGroupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>161</width>
      <height>331</height>
     </rect>
    </property>
    <property name="title">
     <string>System settings</string>
    </property>
    <widget class="QGroupBox" name="functionGroupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>141</width>
       <height>121</height>
      </rect>
     </property>
     <property name="title">
      <string>Function switch</string>
     </property>
     <widget class="QWidget" name="layoutWidget_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>124</width>
        <height>91</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="functionVerticalLayout">
       <item>
        <widget class="QCheckBox" name="faceTrackerCheckBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Face tracking</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="faceRecognizerCheckBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>数据库未完成初始化，请检查</string>
         </property>
         <property name="text">
          <string>Face recognition</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="panalarmCheckBox">
         <property name="text">
          <string>Alarm system</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="webcamGroupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>141</width>
       <height>101</height>
      </rect>
     </property>
     <property name="title">
      <string>Image capture</string>
     </property>
     <widget class="QWidget" name="layoutWidget_4">
      <property name="geometry">
       <rect>
        <x>1</x>
        <y>21</y>
        <width>151</width>
        <height>71</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="webcamVerticalLayout">
       <item>
        <widget class="QCheckBox" name="useExternalCameraCheckBox">
         <property name="text">
          <string>External camera</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="startWebcamButton">
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Turn on the camera</string>
         </property>
         <property name="autoDefault">
          <bool>false</bool>
         </property>
         <property name="default">
          <bool>false</bool>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="dbGroupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>141</width>
       <height>57</height>
      </rect>
     </property>
     <property name="title">
      <string>Database</string>
     </property>
     <widget class="QPushButton" name="initDbButton">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>119</width>
        <height>31</height>
       </rect>
      </property>
      <property name="cursor">
       <cursorShape>PointingHandCursor</cursorShape>
      </property>
      <property name="text">
       <string>Check</string>
      </property>
      <property name="autoDefault">
       <bool>false</bool>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="personalizedSettingsGroupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>181</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>Personalize</string>
    </property>
    <widget class="QGroupBox" name="alarmWaysGroupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>171</width>
       <height>121</height>
      </rect>
     </property>
     <property name="title">
      <string>Alarm method</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="layoutWidget_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>168</width>
        <height>91</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="alarmWaysVerticalLayout">
       <item>
        <widget class="QCheckBox" name="bellCheckBox">
         <property name="text">
          <string>Device sounds</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="telegramBotPushCheckBox">
         <property name="text">
          <string>TelegramBot</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="telegramBotSettingsButton">
         <property name="text">
          <string>Telegram Settings</string>
         </property>
         <property name="autoDefault">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="supportGroupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>171</width>
       <height>81</height>
      </rect>
     </property>
     <property name="title">
      <string>Help and Support</string>
     </property>
     <widget class="QWidget" name="layoutWidget_5">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>20</y>
        <width>165</width>
        <height>66</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="supportVerticalLayout">
       <item>
        <widget class="QPushButton" name="viewGithubRepoButton">
         <property name="text">
          <string>Github repository</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="contactDeveloperButton">
         <property name="text">
          <string>Contact the developer</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="logGroupBox">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>410</y>
      <width>441</width>
      <height>181</height>
     </rect>
    </property>
    <property name="title">
     <string>System log</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <widget class="QTextEdit" name="logTextEdit">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>431</width>
       <height>151</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="realTimeCaptureGroupBox">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>10</y>
      <width>641</width>
      <height>581</height>
     </rect>
    </property>
    <property name="title">
     <string>Camera live view</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <widget class="QLabel" name="realTimeCaptureLabel">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>640</width>
       <height>480</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Camera is not turned on&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>600</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Created by KRC</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
